- Added rotating tips. 5 tips that cycle every 60 seconds with fadeout/fadein

PureKrome opened the pull request:
5 Tips :-
- Type @ then press TAB to auto-complete nicknames
- Type /help to see the list of commands
- Type /rooms to list all available rooms
- Type : then press TAB to auto-complete emoji icons
- You can create your own private rooms. Type /help for more info

Fade Out / Fade In = 2 secs each.

Rotate is set for every 60 seconds.

*\* Apologies for the 2nd commit. I wanted to get my style, right.

PureKrome committed:
- Added rotating tips. 5 tips that cycle every 60 seconds with fadeout/fadein.

PureKrome committed:
- Style cleanup.

davidfowl commented on JabbR/index.htm:
```diff
@@ -43,6 +43,33 @@
+                    });
+
+                    $('#message-instruction').fadeIn(2000, recursiveTimeout());
+                }, 60 * 1000);
```
Make this a constant.

PureKrome replied:
do you mean, the time in milliseconds?

davidfowl replied:
something readable.

davidfowl commented on JabbR/index.htm:
```diff
@@ -43,6 +43,33 @@
+
+            var currentTipIndex = 0;
+
+            function recursiveTimeout() {
```
Make this name more meaningful. cycleTips.

PureKrome committed:
- Renamed the tips array variable.
- Refactored the setTimeout time to a constant.

PureKrome commented:
Updated.

davidfowl commented on JabbR/index.htm:
```diff
@@ -43,6 +43,34 @@
+            var cycleTimeInMilliseconds = 60 * 1000; // 1 minute.
+            var currentTipIndex = 0;
+            
+            function recursiveTimeout() {
```
Change name to cycleMessages

PureKrome replied:
Done.

PureKrome committed:
- Renamed the (javascript) cycle tip variable names.

davidfowl commented on JabbR/index.htm:
```diff
@@ -43,6 +43,34 @@
+            var cycleTimeInMilliseconds = 60 * 1000; // 1 minute.
+            var cycleMessageIndex = 0;
+            
+            function recursiveTimeout() {
```
Change the name of recursiveTimeout to cycleMessages...

PureKrome replied:
Updated. 
(sorry about the slow update -> weekends, kids, no life, QQ....)

PureKrome committed:
- Renamed the setTimeout function.
- Renamed the messages array.

davidfowl commented:
Why is this code in index.htm?

PureKrome commented:
3 reasons.
1. The message div is located in this file.
2. it's tiny.
3. i didn't know of any other js file where it would be appropriate to place it, in. Having it in a seperate js file is preferable because of caching.

I would prefer to have it in a js file. Any preference? chat.ui.js?

davidfowl commented:
Yes move it to the ui file :).

PureKrome committed:
- Moved the message cycling javascript out of the index.htm and into Chat.ui.js
- CycleMessages are now started when the document has finished loading.

PureKrome commented:
Ok. done.

few points to note:
1. I'm kick starting the function logic by calling the jquery `ready` logic. I'm not 100% sure if this is the best place to do it, so I'm happy to move this starting call to a more appropriate place (i'm just not 100% familiar with the js jabbr code).

```
// Start cycling the messages once the document has finished loading.
$(document).ready(cycleMessages());
```
1. Even though i've added a variable to define the cycle time in milliseconds, I'm not sure if that's an overkill. So i'm also happy to replace this..

`var cycleTimeInMilliseconds = 60 * 1000; // 1 minute.
 $('#message-instruction').fadeIn(2000, cycleMessages());
}, cycleTimeInMilliseconds);`

with this

`$('#message-instruction').fadeIn(2000, cycleMessages());
}, 60000);`

davidfowl commented on JabbR/Chat.ui.js:
```diff
@@ -654,6 +654,35 @@
+    };
+
+    // Start cycling the messages once the document has finished loading.
+    $(document).ready(cycleMessages());
```
Yea this is all wrong and doesnt follow the pattern we use today. Kick it off in ui.initialize.

PureKrome replied:
Updated. (added to the bottom of the ui.init.  I didn't know that function existed. Awesome!)

PureKrome committed:
- Remove the $(document).ready and moved the start of the initialising to
  ui.initialize function. (Cheers DFowler)
- BugFix: a function calla was getting invoked immediately. Refactored to be
  a function reference instead. (Cheers redsquare)
